helixbass/tsc-rust
# Parse + bind generic interface declaration

In this PR:
- parse and bind an interface declaration with a simple generic type parameter and a type annotation with a simple generic type argument

To test:
If you run `cargo run tmp.tsx` against a source file containing eg:
```
interface Foo<T> {
  bar: T
}

const a: Foo<number> = {
  bar: true,
}
```
then you should see `post-binding:` debug-logging where the `InterfaceDeclaration`'s `type_parameters` contains a `TypeParameterDeclaration` whose `name` is `"T"`, and then in the `type_` on the `VariableDeclaration` the `type_arguments` contains a `KeywordTypeNode` of `kind: NumberKeyword` (and these should have their `parent` references initialized)

Based on `multiple-checker-modules`
